internal/coreinternal/attraction/coercer.go

```diff
@@ -0,0 +1,69 @@
+package attraction
+
+import (
+	"regexp"
+	"strconv"
+
+	"go.opentelemetry.io/collector/model/pdata"
+)
+
+var num = regexp.MustCompile(`^(\d+)(?:\.\d+)?$`)
+var dub = regexp.MustCompile(`^(\d+(?:\.\d+)?)$`)
+
+func coerceValue(to string, v pdata.AttributeValue) {
+	switch to {
+	case "string":
+		switch v.Type().String() {
+		case "STRING":
+		default:
+			v.SetStringVal(v.AsString())
+		}
+	case "int":
+		switch v.Type().String() {
+		case "INT":
+		case "DOUBLE":
+			v.SetIntVal(int64(v.DoubleVal()))
+		case "BOOL":
+			if v.BoolVal() {
+				v.SetIntVal(1)
+			} else {
+				v.SetIntVal(0)
+			}
+		case "STRING":
+			s := v.StringVal()
+			n := num.FindStringSubmatch(s)
+			if n != nil {
+				intVal, _ := strconv.Atoi(n[1])
+				v.SetIntVal(int64(intVal))
+			} else {
+				v.SetIntVal(int64(0))
+			}
+		default:
+			v.SetIntVal(int64(0))
+		}
+	case "double":
+		switch v.Type().String() {
+		case "INT":
+			v.SetDoubleVal(float64(v.IntVal()))
+		case "DOUBLE":
+		case "BOOL":
+			if v.BoolVal() {
+				v.SetDoubleVal(1)
+			} else {
+				v.SetDoubleVal(0)
+			}
+		case "STRING":
+			s := v.StringVal()
+			n := dub.FindStringSubmatch(s)
+			if n != nil {
```

Comment:
This will not work with the scientific format (e.g. 1.23E+10). I am not sure if it's necessary and instead `ParseFloat` error could be used to decide if conversion was successful?

Reply:
That sounds sensible